src/main/java/org/prebid/cache/handlers/storage/StorageMetricsRecorder.java

```diff
@@ -0,0 +1,63 @@
+package org.prebid.cache.handlers.storage;
+
+import lombok.AllArgsConstructor;
+import org.prebid.cache.builders.PrebidServerResponseBuilder;
+import org.prebid.cache.exceptions.BadRequestException;
+import org.prebid.cache.exceptions.DuplicateKeyException;
+import org.prebid.cache.exceptions.RepositoryException;
+import org.prebid.cache.exceptions.ResourceNotFoundException;
+import org.prebid.cache.metrics.MeasurementTag;
+import org.prebid.cache.metrics.MetricsRecorder;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.reactive.function.server.ServerRequest;
+import org.springframework.web.reactive.function.server.ServerResponse;
+import reactor.core.publisher.Mono;
+
+import java.util.concurrent.TimeoutException;
+
+@AllArgsConstructor
+public class StorageMetricsRecorder {
+
+    private final PrebidServerResponseBuilder responseBuilder;
+    private final MetricsRecorder metricsRecorder;
+    private final String prefix;
+
+    public Mono<ServerResponse> handleErrorMetrics(final Throwable error, final ServerRequest request) {
+        if (error instanceof DuplicateKeyException) {
+            recordMetric(MeasurementTag.ERROR_DUPLICATE_KEY);
+        } else if (error instanceof RepositoryException) {
+            recordMetric(MeasurementTag.ERROR_DB);
+        } else if (error instanceof ResourceNotFoundException || error instanceof BadRequestException) {
+            recordMetric(MeasurementTag.ERROR_BAD_REQUEST);
+        } else if (error instanceof TimeoutException) {
+            recordMetric(MeasurementTag.ERROR_TIMED_OUT);
+        } else {
+            recordMetric(MeasurementTag.ERROR_UNKNOWN);
+        }
+
+        return responseBuilder.error(Mono.just(error), request)
+                .doOnNext(response -> handleErrorStatusCodes(request, response));
```

Comment:
I don't like an idea to "hide" response construction in some cases inside `StorageMetricsRecorder`